python/test/test_polars.py

```diff
@@ -833,7 +833,7 @@ def test_float_sum_with_unlimited_reorderable_partitions():
     # sum of income per region, add noise with scale of 1.0
     pl = pytest.importorskip('polars')
     plan = lf.group_by("region").agg([
-        pl.col("income").dp.sum(bounds=(1_000, 100_000), scale=1.0)
+        pl.col("income").dp.sum(bounds=(1_000.0, 100_000.0), scale=1.0)
```

Comment:
I'd hope the user could still supply ints for bounds and if necessary it would cast to float internally?

Reply:
This is introducing a cast based on the type of the bounds to establish the dtype.